### ISSUE TITLE
expo start --dev-client QR code URL is incorrect

### ISSUE THREAD

**Issue by Douglas-Hsieh** [expo/expo-cli#4020]
### Summary

The $ expo start --dev-client QR code URL for me is wrong.
The URL is "exp://com.username.myapp//expo-development-client/?url=http%3A%2F%2F192.168.1.118%3A8081" when it should be "exp://192.168.1.118:8081"

### Environment

  Expo CLI 4.13.0 environment info:
    System:
      OS: Linux 5.3 Pop!_OS 19.10
      Shell: 5.0.3 - /bin/bash
    Binaries:
      Node: 12.16.1 - ~/.nvm/versions/node/v12.16.1/bin/node
      Yarn: 1.22.5 - /usr/bin/yarn
      npm: 7.13.0 - ~/.nvm/versions/node/v12.16.1/bin/npm
    npmPackages:
      expo: ~43.0.2 => 43.0.3 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.3 => 0.64.3 
      react-native-web: 0.17.1 => 0.17.1 
    npmGlobalPackages:
      expo-cli: 4.13.0
    Expo Workflow: bare

### Please specify your device/emulator/simulator platform, model and version

Physical device: Samsung Galaxy S10

### Error output

Uncaught Error: java.net.UnknownHostException: Unable to resolve host "com.username.myapp": No address associated with hostname

### Reproducible demo or steps to reproduce from a blank project

1. expo init my-app
2. I chose blank typescript managed workflow
3. $ expo run:android
4. I get prompted: "What would you like your Android package name to be?". I choose "com.username.myapp"
5. I get prompted:
"Port 8081 is running my-app in another window
  /home/douglas/projects/my-app (pid 25133)
? Use port 8082 instead?"
I choose no.
6. $ expo start --dev-client
7. My android emulator downloads the bundle and works when I press 'a'
8. My physical android device scans the QR code and opens "exp://com.username.myapp//expo-development-client/?url=http%3A%2F%2F192.168.1.118%3A8081" with the Expo Go app, and fails.
9. On my physical device, I manually edit the link to be "exp://192.168.1.118:8081", copy it, and launch it, it works!

**Comment by alanzheng88** [comment]
@Douglas-Hsieh if you try `expo start -c` does it fix it for you?

**Comment by Douglas-Hsieh** [comment]
@alanzheng88 `expo start -c --dev-client` didn't work for me.

I should add that `expo start` works for me.

**Comment by EvanBacon** [comment]
`exp://` corresponds to the Expo Go app, if you built a custom development client (`expo run:android`) then the URL should correspond to that client, i.e. not Expo Go's `exp://`. If you want the `exp://` URL, then use `expo start` without the dev client flag.

**Comment by joshuaoyerinde** [comment]
**### am having the issue pasted below on expo i installed it on mac**

gyp WARN install got an error, rolling back install
gyp WARN install got an error, rolling back install
gyp ERR! configure error 
gyp ERR! stack Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/expo-cli/node_modules/webpack-dev-server/node_modules/fsevents/.node-gyp'
gyp ERR! System Darwin 20.4.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/expo-cli/node_modules/webpack-dev-server/node_modules/fsevents
gyp ERR! node -v v14.18.1
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok

**Comment by Douglas-Hsieh** [comment]
@EvanBacon My bad, I didn't realize that the development client was an app launcher (similar to Expo Go) that I had to install on my physical device. I now know that there is a QR code for downloading the dev client, and a QR code for downloading the bundle.

When I use the `eas build --profile development --platform android` workflow, it gave me a QR code for my physical device to go to and download the client.

When I use the `expo run:android` workflow, it builds the runtime locally, installs the client on my emulator, and launches the app. 
It also gives me a QR code, but I'm not sure what it's for. Is it supposed to be QR code for downloading the client? (it doesn't work)

It's probably documentation I've missed, but I don't know how to download the dev client using `expo run:android` with a physical device.
I don't see it described in the getting started section of development builds
https://docs.expo.dev/development/getting-started/

**Comment by tcdavis** [comment]
https://docs.expo.dev/development/build/#locally-with-xcode-and-android-studio

You can call (expo run:android -d) to install the build on a connected physical device.

**Comment by Douglas-Hsieh** [comment]
@joshuaoyerinde You might get more attention if you posted this as it's own issue.

It looks like you did some sort of `npm install <package>` / `expo install <package>` command and it gave you a permission denied error? Try `sudo npm install <package>` / `sudo expo install <package>`

**Comment by Douglas-Hsieh** [comment]
@tcdavis Ah, perfect! Thank you

**Comment by dmgcoding** [comment]
yarn run expo start works for me. yarn start didn't work.

**Comment by Wildanzr** [comment]
TLDR; you can swith mode between **development build** and **expo go** by pressing `s`

so this is part of `eas build`, which means you have already created a development build. When you have done this build, you'll get a QR Code to download the bundle. 
You need to download and install your bundle to continue open it. The easiest way to open it without install your bundle is you can swith from between development build to expo go mode.
![image](https://github.com/expo/expo-cli/assets/64595143/e5827462-8821-4ed0-a3fa-c49e90031cb1)
